lib/improver/tests/spotdata_new/spotdata_new/test_NeighbourSelection.py

```diff
@@ -0,0 +1,641 @@
+        self.assertArrayEqual(site_coords_out[0:2], site_coords[0:2])
+        self.assertArrayEqual(out_x, x_points[0:2])
+        self.assertArrayEqual(out_y, y_points[0:2])
+
+        msg = "1 spot sites fall outside the grid"
+        self.assertTrue(any([msg in str(warning) for warning in warning_list]))
+        assert issubclass(warning_list[0].category, UserWarning)
+
+
+class Test_get_nearest_indices(Test_NeighbourSelection):
+
+    """Test function wrapping iris functionality to get nearest grid point
+    indices to arbitrary coordinates."""
+
+    def test_basic(self):
+        """Test that expected coordinates are returned."""
+
+        plugin = NeighbourSelection()
+
+        x_points = np.array([site['projection_x_coordinate']
+                             for site in self.region_sites])
+        y_points = np.array([site['projection_y_coordinate']
+                             for site in self.region_sites])
+        site_coords = np.stack((x_points, y_points), axis=1)
+
+        expected = [[2, 4]]
+        result = plugin.get_nearest_indices(site_coords,
+                                            self.region_orography)
+        self.assertArrayEqual(result, expected)
+
+
+class Test_geocentric_cartesian(Test_NeighbourSelection):
+
+    """Test conversion of global coordinates to geocentric cartesians. In  this
+    coordinate system, x and y are in the equitorial plane, and z is towards
+    the poles."""
+
+    def test_basic(self):
+        """Test a (0, 0) coordinate conversion to geocentric cartesian. This is
+        expected to give an x coordinate which is the semi-major axis of the
+        globe defined in the global coordinate system."""
+
+        plugin = NeighbourSelection()
+        x_points = np.array([0])
+        y_points = np.array([0])
+        result = plugin.geocentric_cartesian(self.global_orography,
+                                             x_points, y_points)
+        radius = self.global_orography.coord_system().semi_major_axis
+        expected = [[radius, 0, 0]]
+        self.assertArrayAlmostEqual(result, expected)
+
+    def test_north_pole(self):
+        """Test a (0, 90) coordinate conversion to geocentric cartesian, this
+        being the north pole. This is expected to give an x coordinate which 0
+        and a z coordinate equivalent to the semi-major axis of the globe
+        defined in the global coordinate system."""
+
+        plugin = NeighbourSelection()
+        x_points = np.array([0])
+        y_points = np.array([90])
+        result = plugin.geocentric_cartesian(self.global_orography,
+                                             x_points, y_points)
+        radius = self.global_orography.coord_system().semi_major_axis
+        expected = [[0, 0, radius]]
+        self.assertArrayAlmostEqual(result, expected)
+
+    def test_45_degrees_latitude(self):
+        """Test a (0, 45) coordinate conversion to geocentric cartesian. In
```

Comment:
Not sure we need all these tests for what is basically a wrapper for a standard cartopy function

Reply:
Yeah. I wondered about that, but I wanted to be sure the functionality was working as I expected. It does mean if cartopy changes underneath us we will notice, so if it's okay I'd like to leave them.